app/static/src/app/store/run/mutations.ts

```diff
@@ -122,5 +129,35 @@ export const mutations: MutationTree<RunState> = {
         if (paramSet) {
             paramSet.hidden = !paramSet.hidden;
         }
+    },
+
+    [RunMutation.SaveParameterDisplayName](state: RunState, payload: ParameterSetNames) {
+        const isDuplicateDisplayName = state.parameterSets.find((set: ParameterSet) => {
+            return set.displayName === payload.newDisplayName;
+        });
+        const isSpecialSetName = payload.newDisplayName.match(/^ *set \d+ *$/gi);
+        const paramSet = state.parameterSets.find((set: ParameterSet) => set.name === payload.parameterSetName);
+        if (paramSet) {
+            const isParamSetName = payload.newDisplayName === paramSet.name;
+            const isParamSetDisplayName = payload.newDisplayName === paramSet.displayName;
+            if (isDuplicateDisplayName && !isParamSetName && !isParamSetDisplayName) {
+                paramSet.isDisplayNameError = true;
+                paramSet.displayNameErrorMsg = "Name already exists";
+            } else if (isSpecialSetName && !isParamSetName) {
+                paramSet.isDisplayNameError = true;
+                paramSet.displayNameErrorMsg = `Set [number] combination is reserved for default set names.
```

Comment:
I wonder about rephrasing to say "Set 2 (or any Set [number] combination)..."

Reply:
done